pulsar-common/src/main/java/com/yahoo/pulsar/common/util/collections/ConcurrentLongPairSet.java

```diff
@@ -528,4 +529,23 @@ public int compareTo(LongPair o) {
             }
         }
     }
+    
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append('{');
+        final AtomicBoolean first = new AtomicBoolean(true);
```

Comment:
but then we need to check empty-map before doing `sb.setLength(sb.length() - 2);` and `toString()` is not synchronized method.